linkerd/core/src/main/scala/io/buoyant/linkerd/RequestActiveTracer.scala

```diff
@@ -0,0 +1,263 @@
+      |dtab resolution:
+      |${dtabResolution.map("  " + _).mkString("\n")}
+    """.stripMargin
+    ).mkString("\n").trim.concat("\n")
+  }
+
+  /**
+   * Returns a list of [[PathNode]] that represents a delegate tree path that identifies how a
+   * router binds a service name.
+   *
+   * @param dTree DelegateTree with a bound name.
+   * @param nodes an accumulator of [[PathNode]] that reveal the path to a client name.
+   * @param clientName the client path name this method is searching for in a [[DelegateTree]]
+   * @return list of [[PathNode]]. An empty list if no path was found.
+   */
+  private[this] def formatDelegation(
+    dTree: DelegateTree[Name.Bound],
+    nodes: List[PathNode],
+    clientName: Path
+  ): List[PathNode] = {
+
+    //walk the delegate tree to find the path of the clientName path.
+    dTree match {
+
+      //when we reach a DelegateTree.Leaf we may have found a path.
+      case l@Leaf(leafPath, dentry, bound) if bound.id == clientName =>
+
+        // We need to check if the last node in 'nodes' is of type DelegationTree.Transformation.
+        // If so, we need to switch the transformation's dentry with
+        // the current leaf's dentry. We do this to make sure that the node list is more readable
+        // when it is added to the request active tracer's response.
+        val finalPath = nodes.lastOption.collect {
+          case PathNode(nodePath, nodeDentry, _: Transformation[_]) => // check if is transformation
+
+            // Switch dentries
+            val transformerNode = PathNode(nodePath, dentry.show)
+            val leafNode = PathNode(leafPath.show, nodeDentry)
+
+            nodes.dropRight(1) :+ transformerNode :+ leafNode
+          case _ =>
+            // If the last node is not a transformation, it's safe to add the leaf node
+            // as the last element of nodes
+            val leafNode = PathNode(leafPath.show, dentry.show, l)
+            nodes :+ leafNode
+        }
+        finalPath.getOrElse(List.empty) // otherwise return  an empty list indicating there is no path.
+      case t@Transformation(path, name, _, remainingTree) =>
+        formatDelegation(remainingTree, nodes :+ PathNode(path.show, name, t), clientName)
+      case d@Delegate(path, dentry, remainingTree) =>
+        formatDelegation(remainingTree, nodes :+ PathNode(path.show, dentry.show, d), clientName)
+      case u@Union(path, dentry, remainingWeightedTrees@_*) =>
+        remainingWeightedTrees.map { wd =>
+          formatDelegation(wd.tree, nodes :+ PathNode(path.show, dentry.show, u), clientName)
+        }.find(!_.isEmpty).toList.flatten
+      case a@Alt(path, dentry, remainingTrees@_*) =>
+        remainingTrees.map { d =>
+          formatDelegation(d, nodes :+ PathNode(path.show, dentry.show, a), clientName)
+        }.find(!_.isEmpty).toList.flatten
+      case _ => List.empty
+    }
+  }
+
+  private def getRequestTraceResponse(resp: Response, stopwatch: Stopwatch.Elapsed) = {
```

Comment:
try to be consistent about `private` vs `private[this]` unless there's a reason this method needs to be accessed from another instance.  I prefer using the smallest visibility scope necessary.